docs/it/senior.md

```diff
@@ -0,0 +1,400 @@
+
+Non ci sono soluzioni corrette o sbagliate, ma solo compromessi differenti. Ed è fondamentale comprendere questi compromessi e prevederne l’effetto, sia nel breve che nel lungo termine. La domanda da farsi è "quale compromesso è quello migliore per gli obiettivi di business presenti e futuri"?
+
+Raggiungere questo livello di maturitá è tutto fuorché semplice. Sicuramente necessita di una significativa quantità di tempo investito nello sviluppare prodotti software, adottando diverse pratiche, diversi linguaggi di programmazione, framework, metodologie e così via.
+
+Potrai ampliare le tue conoscenze solamente se saprai riconoscere che ogni scelta tecnica comporta una serie di compromessi. Ci saranno sempre aspetti positivi e negativi.
+
+Purtroppo non esiste una formula magica per la tecnologia!
+
+Non esiste un singolo linguaggio o framework che possa essere utilizzato per costruire qualsiasi tipo di applicazione in modo ottimale! Se hai una visione sufficientemente completa del panorama tecnologico, dovresti essere in grado di capire quale insieme di tecnologie è il piú adatto per affrontare un problema specifico. Talvolta questo significa essere in grado di mettere da parte le nostre tecnologie preferite o quelle che fanno parte delle nostre competenze primarie e che ci fanno sentire piú produttivi a livello individuale.
+
+Ovviamente, non sempre possiamo avere una visione cristallina di ogni problema. Spesso bisogna confermare le proprie assunzioni con degli esperimenti ad hoc ed essere pronti ad essere smentiti.
+
+Bisogna trovare il modo di lasciare margine per commettere errori, imparare dai fallimenti e provare approcci alternativi. La strada per il successo raramente è una linea retta…
+
+## Andare a caccia di bug
+
+È risaputo che il software privo di bug non esiste, quindi non proviamo neanche a pensare che uno sviluppatore senior possa scrivere codice senza bug!
+
+Benché i bug sono sempre dietro l'angolo, uno sviluppatore senior deve sapere cosa fare per ridurre al minimo il rischio di creare dei bug. Analogamente lo sviluppatore senior deve sapere come individuare i bug il più presto possibile, possibilmente prima che lo facciano gli utenti.
+
+Questo comporta sviluppare tutta una serie di competenze quali: comprendere ed approfondire i resoconti degli utenti, scrivere diversi tipi di test (unitari, di integrazione, end-to-end), individuare ed analizzare i casi limite, strutturare i log in modo che siano più facili da leggere e ricercare in caso di errori, riuscire a leggere ed interpretare varie metriche e dashboard.
+
+Un senior dovrebbe essere in grado di fare una grande differenza e di aiutare l’intero team a scrivere un software migliore, illustrando ed educando alle migliori pratiche per il test del software.
```

Comment:
La butto lì: linkone al capitolo sul testing? @Cadienvan

Reply:
@nicolaerario ci potrebbe stare!

Reply:
C'é un modo ben preciso per creare link tra capitoli?

Reply:
Metodo classico dei link con path relativo (vscode dovrebbe dare anche una mano).
Direi: `[testo del link](testing.md)` , dovrebbe andar bene;
Dovessimo linkare una risorsa in diversa lingua (ad esempio un link ad una pagina che, per assurdo, abbiamo solo in inglese), sarebbe: `[testo del link](../en/testing.md)`